src/passes/Outlining.cpp

```diff
@@ -304,6 +304,13 @@ struct Outlining : public Pass {
     // Position the outlined functions first in the functions vector to make
     // the outlining lit tests far more readable.
     moveOutlinedFunctions(module, substrings.size());
+
+    // Because we visit control flow in an odd order, IRBuilder is not able to
+    // properly track branches, so it may not have finalized blocks with the
+    // correct types. ReFinalize now to fix any issues.
+    PassRunner runner(getPassRunner());
+    runner.add(std::make_unique<ReFinalize>());
+    runner.run();
```

Comment:
This comment sort of suggests to me that Outlining should not be using IRBuilder...? What does "an odd order" mean?

lgtm otherwise

Reply:
Outlining visits control flow expressions in a breadth-first order rather than doing a normal postorder traversal. So it visits every top-level expression in a function body, ends the function, then visits the contents of every block it saw while traversing those top-level expressions, etc. Another way to look at it is that Outlining uses IRBuilder to build one control flow structure at a time rather than a whole function at at a time.

Reply:
This "odd" traversal should be well documented in passes/stringify-walker.h.

Reply:
Ok, sounds good. Maybe add a link to there?